db/db_impl/db_impl.cc

```diff
@@ -2042,6 +2042,50 @@ Status DBImpl::GetEntity(const ReadOptions& _read_options,
   return GetImpl(read_options, key, get_impl_options);
 }
 
+void DBImpl::GetEntity(const ReadOptions& _read_options, const Slice& key,
+                       PinnableAttributeGroups& result) {
```

Comment:
Then we may have no place to return Status (e.g. `PinnableAttributeGroups` is null). I think I can change this to return `Status` while `PinnableAttributeGroups` also contains `Status` per column family.

If input is invalid from the beginning, (e.g. result is null or read_option is not kGetEntity, etc.) we can return `Status::InvalidArgument`, otherwise `Status::OK()`.